controller/uninstall_controller.go

```diff
@@ -0,0 +1,329 @@
+package controller
+
+import (
+	"errors"
+	"time"
+
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/kubernetes/pkg/controller"
+
+	"github.com/Sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers/apps/v1beta2"
+
+	"github.com/rancher/longhorn-manager/datastore"
+	lhinformers "github.com/rancher/longhorn-manager/k8s/pkg/client/informers/externalversions/longhorn/v1alpha1"
+	"github.com/rancher/longhorn-manager/types"
+)
+
+const (
+	GracePeriod = 30 * time.Second
```

Comment:
30 seconds is not enough for deleting a volume or daemonset of engine images.

We can customize it later, but for now, let's try 90 seconds first.